### DISCUSSION TITLE
SSH access to the repository when using the "GIT Repository" server type

### DISCUSSION THREAD

**Discussion by pentastich** [GetPublii/Publii#1381] [General] [answered]
The 0.44.xxx version of Publii introduced "GIT Repository" as a server type. When I go to set one up, I'm prompted to supply a URL, branch, Username, and "Password / Token".

I've used GIT for years, and normally I set up a repository and access it using SSH public keys (see https://git-scm.com/book/en/v2/Git-on-the-Server-Setting-Up-the-Server). For example, to access Publii with SSH, I would use the string "git@github.com:GetPublii/Publii.git".

Can Publii access repositories this way and, if so, how?

**Comment by dziudek** [comment, marked as answer]
Git repositories via SSH AFAIK are not supported in the library used in Publii (isomorphic-git). And the alternative libraries which probably supports this way are not considered by us as they are native modules which are very problematic to maintain and also due licensing conflicts. So at this moment there is no way to solve this issue.

**Reply by pentastich** [reply]
That makes sense. Since SSH is such a common way to access GIT remotely, I suggest adding something to the documentation noting the specific URL forms and authentication mechanisms that are supported. That would help avoid questions like mine!

**Comment by dziudek** [comment]
I have one idea - maybe to consider in the future - use https://www.npmjs.com/package/simple-git as alternative option which can be enabled under advanced settings - then if somebody has git installed, we can use them instead of JS library. 

I will check this concept in the nearest days.

It would be a solution for more advanced users

**Reply by pentastich** [reply]
That would be perfect for me. I plan to run the final site off of github, but I also have a local server that I use as a playpen.

**Reply by dziudek** [reply]
@pentastich - yes, I see many benefits - there will be wider range of git features available. The current solution will be left as default which not requires dependencies, but more advanced users will be able to use their own git instances.

**Reply by KristijanK** [reply]
Same here would love to have run the website on a local server without GitHub just with a bare git repo

**Reply by KristijanK** [reply]
Hey guys, any plans or drafts regarding this topic? I would love to support anyhow because I would love to have that kind of feature. Any starting point so I could try to dig deeper into the topic, at least to give it a try. Thanks so far...
